File: backend/src/main/java/com/laboQuimica/kalium/controller/ChatController.java
package com.laboQuimica.kalium.controller;

import com.laboQuimica.kalium.service.GeminiService;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.*;

@RestController
@RequestMapping("/api/chat")
@CrossOrigin(origins = "*")
public class ChatController {

    private final GeminiService geminiService;

    @Autowired
    public ChatController(GeminiService geminiService) {
        this.geminiService = geminiService;
    }

    @PostMapping
    public ResponseEntity<?> chat(@RequestBody String message) {
        try {
            if (message == null || message.trim().isEmpty()) {
                return ResponseEntity.badRequest().body("El mensaje no puede estar vacío");
            }
            
            String response = geminiService.generateResponse(message);
            return ResponseEntity.ok().body(response);
            
        } catch (Exception e) {
            return ResponseEntity.internalServerError()
                .body("Error al procesar el mensaje: " + e.getMessage());
        }
    }
}

File: backend/controllers/chatController.js
const { generateResponse } = require('../services/geminiService');

const chatWithAI = async (req, res) => {
  try {
    const { message } = req.body;
    
    if (!message) {
      return res.status(400).json({ error: 'El mensaje es requerido' });
    }

    const response = await generateResponse(message);
    res.json({ response });
  } catch (error) {
    console.error('Error en el controlador de chat:', error);
    res.status(500).json({ error: 'Error al procesar la solicitud' });
  }
};

module.exports = { chatWithAI };
